rs/boundary_node/rate_limit_canister/interface.did

```diff
@@ -0,0 +1,104 @@
+type RuleId = text;             // Unique identifier for each rule
+type SchemaVersion = nat64;     // Version of the schema for encoding/decoding the rules
+type IncidentId = text;         // Unique identifier for each incident
+
+
+// Input structure for defining a rule with mandatory fields within a config
+type InputRule = record {
+  incident_id: IncidentId; // Identifier for the incident, to which the rule is related
+  rule_raw: blob;          // Raw rule data (in binary format), expected to be a valid json object
+  description: text;       // Textual description of the rule
+};
+
+// Output structure for rules
+// Optional fields rule_raw and description may remain hidden while the rule is under confidentiality restrictions
+type OutputRule = record {
+  rule_id: RuleId;                   // Unique identifier for the rule
+  rule_raw: opt blob;                // Raw rule data (in binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;             // Textual description of the rule, none if the rule is currently confidential
+};
+
+type OutputConfig = record {
+  schema_version: SchemaVersion; // schema version needed to deserialize the rules
+  rules: vec OutputRule;
+};
+
+// Response structure for returning the requested configuration and associated metadata
+type OutputConfigResponse = record {
+    version: Version;         // Version of the configuration
+    active_since: Timestamp;  // Time when this configuration was added (became active)
+    config: OutputConfig;     // Contains the list of rules
+};
+
+// Verbose details of an individual rule
+// Optional rule_raw and description fields are for restricted publicly viewing access
+type OutputRuleMetadata = record {
+  rule_id: RuleId;                 // Unique identifier for the rule
+  incident_id: IncidentId;         // Identifier for the incident, to which the rule is related
+  rule_raw: opt blob;              // Raw rule data (binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;           // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;     // Timestamp when the rule was disclosed, none if the rule is still confidential
+  added_in_version: Version;       // Version when the rule was added (became active)
+  removed_in_version: opt Version; // Version when the rule was deactivated (removed), none if the rule is still active
+};
+
+type GetRuleByIdResponse = variant {
+  Ok: OutputRuleMetadata;
+  Err: text;
+};
+
+type GetConfigResponse = variant {
+  Ok: OutputConfigResponse;
+  Err: text;
+};
+
+type OverwriteConfigResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesArg = variant {
+  RulesIds: vec RuleId;
+  IncidentIds: vec IncidentId;
```

Comment:
Should it be vec for both? And there is a typo.
```suggestion
  RuleIds: vec RuleId;
  IncidentIds: vec IncidentId;
```

Reply:
> Should it be vec for both?

IMO yes, why not.

> And there is a typo.

yes.